homeassistant/components/sensor/yr.py

```diff
@@ -185,7 +187,8 @@ def try_again(err: str):
                 model = self.data['meta']['model']
                 if '@nextrun' not in model:
                     model = model[0]
-                self._nextrun = dt_util.parse_datetime(model['@nextrun'])
+                self._nextrun = dt_util.parse_datetime(model['@nextrun']) +\
```

Comment:
I hear you, but after _nextrun, we also do not have any valid local data for the current timeperiod. So we will update the sensor and only have a chance to request new date on the next "randomized" min on the hour